NPE in SeoSitemapLinkExternalizer when page is published implicitly

When a page is published implicitly (i.e. when it's child is published but the page itself not), AEM creates on publish a `cq:Page` node without `jcr:content`. This causes NPE when `SeoSitemapLinkExternalizerImpl` tries to adapt page content to LinkHandler.

```
Caused by: java.lang.NullPointerException: Cannot invoke "org.apache.sling.api.adapter.Adaptable.adaptTo(java.lang.Class)" because "adaptable" is null
  at io.wcm.sling.commons.adapter.AdaptTo.notNull(AdaptTo.java:46) [io.wcm.sling.commons:1.6.4]
  at io.wcm.handler.link.impl.SeoSitemapLinkExternalizerImpl.externalizePageLink(SeoSitemapLinkExternalizerImpl.java:140) [io.wcm.handler.link:2.2.8._cp2fm-converted]
  at io.wcm.handler.link.impl.SeoSitemapLinkExternalizerImpl.externalize(SeoSitemapLinkExternalizerImpl.java:113) [io.wcm.handler.link:2.2.8._cp2fm-converted]
  at com.adobe.aem.wcm.seo.impl.sitemap.ExternalizerHelper.externalize(ExternalizerHelper.java:45) [com.adobe.cq.wcm.com.adobe.aem.wcm.seo.impl:1.0.16]
  at com.adobe.aem.wcm.seo.impl.sitemap.PageTreeSitemapGeneratorImpl.getCanonicalUrl(PageTreeSitemapGeneratorImpl.java:241) [com.adobe.cq.wcm.com.adobe.aem.wcm.seo.impl:1.0.16]
  at com.adobe.aem.wcm.seo.impl.sitemap.PageTreeSitemapGeneratorImpl.getAlternateLanguageLinks(PageTreeSitemapGeneratorImpl.java:259) [com.adobe.cq.wcm.com.adobe.aem.wcm.seo.impl:1.0.16]
  at com.adobe.aem.wcm.seo.impl.sitemap.PageTreeSitemapGeneratorImpl.addResource(PageTreeSitemapGeneratorImpl.java:158) [com.adobe.cq.wcm.com.adobe.aem.wcm.seo.impl:1.0.16]
  at org.apache.sling.sitemap.spi.generator.ResourceTreeSitemapGenerator.generate(ResourceTreeSitemapGenerator.java:58) [org.apache.sling.sitemap:1.0.6]
  at org.apache.sling.sitemap.impl.SitemapGeneratorExecutor.generate(SitemapGeneratorExecutor.java:158) [org.apache.sling.sitemap:1.0.6]
  ... 7 common frames omitted
```

Expected result:
* No NPE
* Default SEO externalizer

--- a/changes.xml
+++ b/changes.xml
@@ -24,6 +24,12 @@
     xsi:schemaLocation="http://maven.apache.org/changes/2.0.0 https://maven.apache.org/xsd/changes-2.0.0.xsd">
   <body>
 
+    <release version="2.2.10" date="not-released">
+      <action type="fix" dev="mrozati" issue="31">
+        SeoSitemapExternalizer: fix NPE when page is published implicitly and jcr:content is missing.
+      </action>
+    </release>
+
     <release version="2.2.8" date="2025-07-28">
       <action type="add" dev="henrykuijpers" issue="26">
         I18n: Add Dutch translations.

--- a/src/main/java/io/wcm/handler/link/impl/SeoSitemapLinkExternalizerImpl.java
+++ b/src/main/java/io/wcm/handler/link/impl/SeoSitemapLinkExternalizerImpl.java
@@ -136,7 +136,7 @@ public class SeoSitemapLinkExternalizerImpl implements SitemapLinkExternalizer {
   }
 
   private @Nullable String externalizePageLink(@Nullable Page page) {
-    if (page != null) {
+    if (page != null && page.hasContent()) {
       LinkHandler linkHandler = AdaptTo.notNull(page.getContentResource(), LinkHandler.class);
       String url = linkHandler.get(page).urlMode(UrlModes.FULL_URL).buildUrl();
       if (url != null) {

--- a/src/test/java/io/wcm/handler/link/impl/SeoSitemapLinkExternalizerImplTest.java
+++ b/src/test/java/io/wcm/handler/link/impl/SeoSitemapLinkExternalizerImplTest.java
@@ -113,6 +113,23 @@ class SeoSitemapLinkExternalizerImplTest {
   }
 
   @Test
+  void testExternalizeResource_Page_NoContentResource() {
+    MockContextAwareConfig.writeConfiguration(context, ROOTPATH_CONTENT, SiteConfig.class,
+        "siteUrl", "https://myhost");
+
+    // When a page is published implicitly i.e. when it's child is published but the page itself not,
+    // AEM creates a page node without jcr:content
+    Resource resource = context.create().resource(ROOTPATH_CONTENT + "/page2", "jcr:primaryType", "cq:Page");
+
+    when(aemSitemapLinkExternalizer.externalize(resource)).thenReturn("defaultResult");
+
+    String result = underTest.externalize(resource);
+    assertEquals("defaultResult", result);
+
+    verifyNoMoreInteractions(aemSitemapLinkExternalizer);
+  }
+
+  @Test
   void testExternalizeResource_NotPage() {
     Resource resource = context.create().resource("/content/resource1");
 

